eppo_core/src/event_ingestion/event_delivery.rs

```diff
@@ -88,3 +98,74 @@ impl EventDelivery {
         Ok(response)
     }
 }
+
+fn ensure_max_event_size(event: &Event) -> Result<(), EventDeliveryError> {
+    let serialized_event = serde_json::to_vec(event)
+        .map_err(|e| JsonSerializationError(e))?;
```

Comment:
this is somewhat wasteful since we're serializing each event and throwing away the output, then later we serialize the entire body again, but this makes the code simpler and doesn't seem worth the tradeoff for optimization